grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -733,7 +736,7 @@ class StockMovementService {
             case "2":
                 return getAddPageItem(requisition, stockMovementItem)
             case "4":
-                return buildPickPageItem(requisitionItem, stockMovementItem.sortOrder, showDetails)
+                return generatePickPageItem(requisitionItem, stockMovementItem.sortOrder, showDetails)
```

Comment:
~+1 on that question~ nvm, I just went to the end of PR, and now I get it. Wouldn't it be better to still have it as a `buildPickPageItem`, but with `generateIfMissing` or even `createPick` (or something like that) flag since you are still building the response but with a few extra steps?

Reply:
I was trying to avoid passing in too many parameter which is why I decided to split this one method in two parts 
1. buildPickPageItem - builds the PickPageItem, simple return of the value without any side effects
2. generatePickPageItem -> creates a new picklist item and calls 1. buildPickPageItem to return this value